packages/contracts/test/contracts/discovery/Controller.ts

```diff
@@ -82,4 +92,48 @@ describe("Controller", () => {
     it("fails to register a project that's already registered in a batch");
     it("fails to register a project that is not ready for listing");
   });
+
+  describe("createBatch", () => {
+    it("creates a batch", async () => {
+      const tx = await controller.registerProject(
+        "My Project",
+        projectToken.address,
+        parseUnits("1000"),
+        parseUnits("2")
+      );
+
+      const event = await findEvent(tx, "ProjectRegistered");
+      const projectAddress = event?.args?.project;
+
+      const project = Project__factory.connect(projectAddress, owner);
+
+      await projectToken
+        .connect(owner)
+        .transfer(project.address, parseUnits("1000"));
+
+      await project.approveByLegal();
+      await project.approveByManager();
+
+      expect(await project.approvedByLegal()).to.equal(true);
+      expect(await project.approvedByManager()).to.equal(true);
+      expect(await project.isReadyForListing()).to.equal(true);
+
+      await expect(controller.createBatch([projectAddress], 4)).to.not.be
+        .reverted;
+    });
+
+    it("reverts if a project is not approved", async () => {
+      const tx = await controller.registerProject(
+        "My Project",
+        alice.address,
+        parseUnits("1000"),
+        parseUnits("2")
+      );
+
+      const event = await findEvent(tx, "RegisterProject");
+      const projectAddress = event?.args?.project;
+
+      await expect(controller.createBatch([projectAddress], 4)).to.be.reverted;
+    });
+  });
```

Comment:
can you add a test for `reverts if a project is already included in a different batch`?